test/mjsonwp/mjsonwp-specs.js

```diff
@@ -217,18 +217,32 @@ describe('MJSONWP', async () => {
         json: {id: 'baz', sessionId: 'lol', value: ['a']}
       });
 
+      // make sure adding the optional 'id' doesn't clobber a route where we
+      // have an actual required 'id'
       await request({
-        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        url: 'http://localhost:8181/wd/hub/session/foo/frame',
         method: 'POST',
         json: {id: 'baz'}
+      });
+    });
+
+    it('should properly handle variable payload for set value call', async () => {
+      await request({
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        method: 'POST',
+        json: {sessionId: 'lol'}
       }).should.eventually.be.rejectedWith("400");
 
-      // make sure adding the optional 'id' doesn't clobber a route where we
-      // have an actual required 'id'
       await request({
-        url: 'http://localhost:8181/wd/hub/session/foo/frame',
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
         method: 'POST',
-        json: {id: 'baz'}
+        json: {sessionId: 'lol', value: ['a'], text: 'bla'}
+      });
+
+      await request({
+        url: 'http://localhost:8181/wd/hub/session/foo/element/bar/value',
+        method: 'POST',
+        json: {sessionId: 'lol', text: 'bla'}
```

Comment:
This is how we parse route arguments:

```javascript
function makeArgs (reqParams, jsonObj, payloadParams) {
  // we want to pass the url parameters to the commands in reverse order
  // since the command will sometimes want to ignore, say, the sessionId
  let urlParams = _.keys(reqParams).reverse();

  // there can be multiple sets of required params, so find the correct one
  let realRequiredParams = payloadParams.required;
  if (_.isArray(_.first(payloadParams.required))) {
    let keys = _.keys(jsonObj);
    for (let params of payloadParams.required) {
      // check if all the required parameters are in the json object
      if (_.without(params, ...keys).length === 0) {
        // we have all the parameters for this set
        realRequiredParams = params;
        break;
      }
    }
  }
  let args = _.flatten(realRequiredParams).map((p) => jsonObj[p]);
  if (payloadParams.optional) {
    args = args.concat(_.flatten(payloadParams.optional).map((p) => jsonObj[p]));
  }
  args = args.concat(urlParams.map((u) => reqParams[u]));
  return args;
}
```

and this is what is expected by FakeDriver:

```javascript
 async setValue (elementId, value) {
    return value;
  }
```

question - how **text** can be retuned here since if it only expects **value** to be set?